include/envoy/network/connection.h

```diff
@@ -233,6 +234,15 @@ class Connection : public Event::DeferredDeletable, public FilterManager {
    * Get the socket options set on this connection.
    */
   virtual const ConnectionSocket::OptionsSharedPtr& socketOptions() const PURE;
+
+  /**
+   * Object on which filters can share data on a per-connection basis.
+   * Only one filter can produce a named data object, but it may be
+   * consumed by many other objects.
+   * @return the per-connection state associated with this connection.
+   */
+  virtual Envoy::RequestInfo::FilterState& perConnectionState() PURE;
```

Comment:
nit: It doesn't thrill me that this is in the `RequestInfo` namespace since this has nothing to do with requests at this point. Can we move to some other sensibly named namespace (or an existing one)?

Reply:
This all came about because I added access logging to TcpProxy, and reused the existing RequestInfo. So the state of things right now is that RequestInfo is used in both http requests and tcp proxy connections. And some of the logging fields are documented as behaving differently in tcp proxy.

I think we can probably leave the code mostly alone, and just come up with a better name than RequestInfo that encompasses both use cases. But names are hard, and I don't have a good idea.

Reply:
@htuch @mattklein123 What about we leave this PR as-is and file an issue to rename the class/namespace for this stuff?

Reply:
+1

Reply:
OK SGTM. I can't immediately think of a better name right now anyway.

Reply:
FlowInfo :)

Reply:
or TransactionInfo 👎

Reply:
Yeah I guess either FlowInfo or StreamInfo would be good and generic enough. +1

Reply:
ConnectionInfo? I think it makes clearer the L4 nature if we're not multiplexing. Streams/flows to me conjure up L7 H2.

Reply:
I think we are find a generic name for both L4/L7? if so StreamInfo SGTM.

Reply:
Oh, I see. Sure, StreamInfo.

Reply:
We could always subclass/typedef for the respective Connection/Request variants.

Reply:
Filed #4500